raiden/transfer/mediated_transfer/initiator_manager.py

```diff
@@ -270,23 +229,36 @@ def handle_cancelpayment(
     return TransitionResult(payment_state, events)
 
 
-def handle_transferrefundcancelroute(
+def handle_failroute(
+    payment_state: InitiatorPaymentState, state_change: ReceiveTransferCancelRoute
+) -> TransitionResult[InitiatorPaymentState]:
+
+    events: List[Event] = list()
+
+    initiator_state = payment_state.initiator_transfers.get(state_change.transfer.lock.secrethash)
+    if initiator_state and can_cancel(initiator_state):
+        cancel_events = cancel_current_route(payment_state, initiator_state)
+        events.extend(cancel_events)
+
+    return TransitionResult(payment_state, events)
+
+
+def handle_transferreroute(
     payment_state: InitiatorPaymentState,
-    state_change: ReceiveTransferRefundCancelRoute,
+    state_change: ActionTransferReroute,
     channelidentifiers_to_channels: Dict[ChannelID, NettingChannelState],
     nodeaddresses_to_networkstates: NodeNetworkStateMap,
     pseudo_random_generator: random.Random,
     block_number: BlockNumber,
 ) -> TransitionResult[InitiatorPaymentState]:
-    initiator_state = payment_state.initiator_transfers.get(state_change.transfer.lock.secrethash)
-    if not initiator_state:
-        return TransitionResult(payment_state, list())
+    events: List[Event] = []
 
-    channel_identifier = initiator_state.channel_identifier
-    channel_state = channelidentifiers_to_channels.get(channel_identifier)
-
-    if not channel_state:
-        return TransitionResult(payment_state, list())
+    try:
+        initiator_state = payment_state.initiator_transfers[state_change.transfer.lock.secrethash]
+        channel_identifier = initiator_state.channel_identifier
+        channel_state = channelidentifiers_to_channels[channel_identifier]
+    except KeyError:
+        return TransitionResult(payment_state, events)
```

Comment:
Why not return 
```suggestion
        return TransitionResult(payment_state, [])
```
here and create `events` below? Would create less local state and thus make reading the code slightly easier.